Sources/ProtobufferBuilder/Extensions/Model+TypeInfo.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  File.swift
+//  
+//
+//  Created by Nityananda on 27.11.20.
+//
+
+import Runtime
+
+extension Message {
+    init(typeInfo: TypeInfo) throws {
+        let name = try typeInfo.compatibleName() + "Message"
+        
+        let properties: [Property]
+        
+        if ParticularType(typeInfo.type).isPrimitive {
+            properties = []
+        } else {
+            properties = typeInfo.properties
+                .enumerated()
+                .compactMap { tuple -> Property? in
+                    let (offset, element) = tuple
+                    do {
+                        let typeInfo = try Runtime.typeInfo(of: element.type)
+                        let postfix = ParticularType(element.type).isPrimitive
+                            ? ""
+                            : "Message"
+                        let typeName = try typeInfo.compatibleName() + postfix
+                        
+                        return Property(
+                            isRepeated: ParticularType(typeInfo.type).isArray,
+                            name: element.name,
+                            typeName: typeName,
+                            uniqueNumber: offset
+                        )
+                    } catch {
+                        print(error)
+                        return nil
```

Comment:
I haven't actually encountered erroneous properties, but I have encountered two types of Errors here:

- `couldNotGetTypeInfo(type: Builtin.Int8, kind: Runtime.Kind.opaque)`
- `couldNotGetTypeInfo(type: Builtin.BridgeObject, kind: Runtime.Kind.opaque)`

I use this mechanism to find an _organic_ end to my tree depth.
The above errors are thrown, if Runtime inspects internal implementations of `Swift.String`, for example.

Sadly, I cannot specialize the catch statement with pattern matching, because Runtime.RuntimeError is internal.